src/psij/executors/batch/test/qlib.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python3
+import json
+import logging
+import tempfile
+from datetime import timedelta, datetime
+from pathlib import Path
+from typing import Dict, Optional, cast
+
+from filelock import FileLock
```

Comment:
I don't know about that. It's my understanding that sqlite3 is barely thread-safe, let alone multi-process safe. And if it was, it would still need some inter-process locking mechanism to ensure that different processes don't both mangle each other's databases, which will likely require something like filelock under the covers, at least when opening a DB. 

[This](https://www.sqlite.org/faq.html#q5), for example, suggests that `fnctl()` locking is used by the sqlite3 implementation, which is what `filelock` uses on unix (see https://py-filelock.readthedocs.io/en/latest/api.html#filelock.UnixFileLock).

Reply:
It is both thread-safe and multi-process safe, it says so in the FAQ you linked to. The advantage (as I see it) isn't that it has some magic secret sauce to be more thread-safe than `filelock`, just that it is probably faster and more intuitive than managing a custom JSON object with (possibly) a bunch of entries, reading it in and writing it out all the time. And it is one less dependency.

But `filelock` isn't a big dependency so I'm happy to keep it.

Reply:
> just that it is probably faster and more intuitive than managing a custom JSON object with (possibly) a bunch of entries

This is a fake queuing system used for testing. To me SQLite seems like too much complexity for the task, but maybe I'm wrong.

Reply:
Yeah, it is like killing an insect with a rocket launcher. We can always change it later if we decide we want to.